plugin/torch/torch_base.cc

```diff
@@ -40,9 +40,11 @@ void TorchState::SetStream(mshadow::Stream<mshadow::cpu>* s) {
 }
 
 #if MXNET_USE_CUDA
+
 template<>
 void TorchState::SetStream(mshadow::Stream<mshadow::gpu>* s) {
-  CudaState()->currentStream = mshadow::Stream<gpu>::GetStream(s);
+  cudaStream_t cs = THCState_getCurrentStream(CudaState());
```

Comment:
Thanks for your reviews.

Oh, I was wrong. Although, we cannot use that because
`void THCState_setStream(THCState *state, THCStream *stream)` won't accept `cudaStream_t` from mshadow::Stream::GetStream.

So I will revert to my first commit https://github.com/dmlc/mxnet/pull/4329/commits/b9a1f7db20633af61644e40876f434c3ea5be2a8 using `THCState_setStream`